packages/perseus/src/renderer-util.ts

```diff
@@ -53,27 +53,13 @@ export function emptyWidgetsFunctional(
             return false;
         }
 
-        let score: PerseusScore | null = null;
-        const userInput = userInputMap[id];
         const scorer = getWidgetScorer(widget.type);
-
-        if (widget.type === "group") {
-            const scores = scoreWidgetsFunctional(
-                widget.options.widgets,
-                Object.keys(widget.options.widgets),
-                userInputMap[id] as UserInputMap,
-                strings,
-                locale,
-            );
-            score = Util.flattenScores(scores);
-        } else if (scorer) {
-            score = scorer(
-                userInput as UserInput,
-                widget.options,
-                strings,
-                locale,
-            );
-        }
+        const score = scorer?.(
```

Comment:
Question: I don't remember seeing a function called like this before. So you can do a null check before calling a function by adding a question mark and a period?

Reply:
Yes.  It's call an ["optional chaining"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) operator.  

With it, you can safely access a property on an object, even if the object is null/undefined at runtime. The result of the call will short-circuit and be `undefined` at runtime. It avoids 2 extra lines of code for an `if (x != null) {` and the closing `}`.